python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py

```diff
@@ -620,6 +620,30 @@ def aten〇adaptive_avg_pool2d〡shape(self: List[int], output_size: List[int])
 def aten〇flatten〇using_ints〡shape(self: List[int], start_dim: int = 0, end_dim: int = -1) -> List[int]:
     return upstream_shape_functions.flatten(self, start_dim, end_dim)
 
+@check_shape_function([
+    Invocation(TensorOfShape(3, 6, 8), 1, [3, 2]),
+    Invocation(TensorOfShape(3, 6, 8), 1, [3, -1]),  # contain one -1 in sizes
+    Invocation(TensorOfShape(3, 6, 8), -1, [2, -1, 2]),  # dim = -1
+    ErrorInvocation(TensorOfShape(3, 6, 8), 1, [-1, -1, 2]), # contain two -1 in sizes
+])
+def aten〇unflatten〇int〡shape(self: List[int], dim: int, sizes: List[int]) -> List[int]:
```

Comment:
I think we can reuse `aten.view`'s shape function:
```
def aten〇unflatten〇int〡shape(self: List[int], dim: int, sizes: List[int]) -> List[int]:
    if dim < 0:
        dim = dim + len(self)
    unflatten_shape: List[int] = [self[dim]]
    unflatten_shape_output = upstream_shape_functions.view(unflatten_shape, sizes)
    shape: List[int] = []
    for i in range(len(self)):
        if i != dim:
            shape.append(self[i])
        else:
            shape = shape + unflatten_shape_output
    return shape
```

Reply:
Good catch!